src/interface/window.c

```diff
@@ -2477,3 +2479,29 @@ void window_update_textbox()
 		window_event_textinput_call(w, gCurrentTextBox.widget_index, gTextBoxInput);
 	}
 }
+
+bool window_is_visible(rct_window* w)
+{
+	// only consider viewports, consider the main window always visible
+	if (w == NULL || w->viewport == NULL || w->classification == WC_MAIN_WINDOW)
+	{
+		// default to previous behavior
+		return true;
+	}
+
+	// start from the window above the current
+	for (rct_window *w_other = (w+1); w_other < RCT2_NEW_WINDOW; w_other++)
+	{
+		// if covered by a higher window, no rendering needed
+		if (w_other->x <= w->x
+			&& w_other->y <= w->y
+			&& w_other->x + w_other->width >= w->x + w->width
+			&& w_other->y + w_other->height >= w->y + w->height)
+		{
+			return false;
+		}
+	}
+
+	// default to previous behavior
```

Comment:
Considering Sawyer himself was a British bloke it'd only be a tribute to him to continue his legacy in the comments 😃